### PR TITLE
MYFACES-4637: Encode Java Records to JSON

### PR DESCRIPTION
**Pull request by volosied**
https://issues.apache.org/jira/browse/MYFACES-4637

Without using a new dependency, this may be the best we can do.

I added a single test case for some basic verification.

### CODE REVIEW

File: impl/src/main/java/org/apache/myfaces/push/Json.java [hunk 1]
```diff
@@ -111,12 +112,45 @@ else if (object instanceof Class<?>)
...
+            String name = field.getName();
+            builder.append(name);
+            builder.append(':');
+            try {
+                encode(field.get(recordObject), builder);
+            } catch (IllegalArgumentException e) {
+                e.printStackTrace();
+            } catch (IllegalAccessException e) {
+                e.printStackTrace();
+            }
```
Comment (volosied) [inline impl/src/main/java/org/apache/myfaces/push/Json.java:143]: I'm not sure what I should do with error handling here.

Comment (melloware) [reply impl/src/main/java/org/apache/myfaces/push/Json.java:143]: I feel like we should throw some sort of unchecked exception here right?

File: impl/src/main/java/org/apache/myfaces/push/Json.java [hunk 2]
```diff
@@ -111,12 +112,45 @@ else if (object instanceof Class<?>)
...
     }
 
+    private static void encodeRecord(Object recordObject, StringBuilder builder) {
+
+        builder.append('{');
+        boolean fieldsFound = false;
+
+        for (Field field : recordObject.getClass().getDeclaredFields())
+        {
+            field.setAccessible(true);
```
Comment (melloware) [inline impl/src/main/java/org/apache/myfaces/push/Json.java:132]: I could be wrong but is this the kind of thing they are removing in JDK 21? Being able to change field access?  I don't follow those JDK reflection updates much but i thought they were removing a lot of things?

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by volosied** [comment]
@melloware  Take a look at the new code -- I used a Record Specific class to get the fields.

https://download.java.net/java/early_access/panama/docs/api/java.base/java/lang/reflect/RecordComponent.html  ( Only found Java 19 javadoc, but this class was added in 16).

**Review by melloware — APPROVED** [review]

**Comment by melloware** [comment]
Slick!

**Pull request merged**